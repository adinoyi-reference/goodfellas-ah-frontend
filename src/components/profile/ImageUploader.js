import React, { Component } from 'react';
import icons from '../../assets/icons.svg';
import { userPlaceholderImage } from '../../mixin';

class ProfileImageUploader extends Component {
  handleChange = (e) => {
    const { imageRead } = this.props;
    const file = e.target.files[0];
    e.preventDefault();
    const reader = new FileReader();
    reader.onloadend = () => {
      imageRead(file, reader.result);
    };
    if (file) reader.readAsDataURL(file);
  }

  render() {
    const { profileImage, canReset, resetImage } = this.props;
    return (
      <div
        className="profile-image profile-image-uploader centralizer"
        style={{ backgroundImage: `linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(${profileImage || userPlaceholderImage})` }}
      >
        {canReset && <span onClick={resetImage} className="profile-image-uploader_cancel hoverable centralizer">x</span>}
        <input type="file" onChange={this.handleChange} id="uploader" accept=".png, .jpg, .jpeg" />
        <label htmlFor="uploader">
          <svg className="icon">
            <use xlinkHref={`${icons}#camera`} />
          </svg>
        </label>
      </div>
    );
  }
}

export default ProfileImageUploader;
