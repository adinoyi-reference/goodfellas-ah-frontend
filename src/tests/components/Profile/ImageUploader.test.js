import React from 'react';
import { shallow } from 'enzyme';
import ImageUploader from '../../../components/profile/ImageUploader';

const setup = () => {
  const props = {
    canReset: false,
    type: 'follower',
    imageRead: jest.fn(),
    resetImage: jest.fn(),
  };

  const enzymeWrapper = shallow(<ImageUploader {...props} />);

  return {
    props,
    enzymeWrapper,
  };
};

describe('components', () => {
  describe('ProfileToolbar component', () => {
    it('should render as expected', () => {
      const { enzymeWrapper } = setup();
      expect(enzymeWrapper).toMatchSnapshot();
    });
  });
});
