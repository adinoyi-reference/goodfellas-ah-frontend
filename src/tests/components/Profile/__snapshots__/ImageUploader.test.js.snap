// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components ProfileToolbar component should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileImageUploader
    canReset={false}
    imageRead={[MockFunction]}
    resetImage={[MockFunction]}
    type="follower"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        false,
        <input
          accept=".png, .jpg, .jpeg"
          id="uploader"
          onChange={[Function]}
          type="file"
        />,
        <label
          htmlFor="uploader"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#camera"
            />
          </svg>
        </label>,
      ],
      "className": "profile-image profile-image-uploader centralizer",
      "style": Object {
        "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png)",
      },
    },
    "ref": null,
    "rendered": Array [
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": ".png, .jpg, .jpeg",
          "id": "uploader",
          "onChange": [Function],
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#camera"
            />
          </svg>,
          "htmlFor": "uploader",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#camera"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#camera",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          false,
          <input
            accept=".png, .jpg, .jpeg"
            id="uploader"
            onChange={[Function]}
            type="file"
          />,
          <label
            htmlFor="uploader"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#camera"
              />
            </svg>
          </label>,
        ],
        "className": "profile-image profile-image-uploader centralizer",
        "style": Object {
          "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png)",
        },
      },
      "ref": null,
      "rendered": Array [
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "accept": ".png, .jpg, .jpeg",
            "id": "uploader",
            "onChange": [Function],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#camera"
              />
            </svg>,
            "htmlFor": "uploader",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#camera"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#camera",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`components ProfileToolbar component should render as expected 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ProfileImageUploader
    canReset={true}
    imageRead={[MockFunction]}
    resetImage={[MockFunction]}
    type="follower"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="profile-image-uploader_cancel hoverable centralizer"
          onClick={[MockFunction]}
        >
          x
        </span>,
        <input
          accept=".png, .jpg, .jpeg"
          id="uploader"
          onChange={[Function]}
          type="file"
        />,
        <label
          htmlFor="uploader"
        >
          <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#camera"
            />
          </svg>
        </label>,
      ],
      "className": "profile-image profile-image-uploader centralizer",
      "style": Object {
        "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png)",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "x",
          "className": "profile-image-uploader_cancel hoverable centralizer",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": "x",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "accept": ".png, .jpg, .jpeg",
          "id": "uploader",
          "onChange": [Function],
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <svg
            className="icon"
          >
            <use
              xlinkHref="icons.svg#camera"
            />
          </svg>,
          "htmlFor": "uploader",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <use
              xlinkHref="icons.svg#camera"
            />,
            "className": "icon",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "xlinkHref": "icons.svg#camera",
            },
            "ref": null,
            "rendered": null,
            "type": "use",
          },
          "type": "svg",
        },
        "type": "label",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="profile-image-uploader_cancel hoverable centralizer"
            onClick={[MockFunction]}
          >
            x
          </span>,
          <input
            accept=".png, .jpg, .jpeg"
            id="uploader"
            onChange={[Function]}
            type="file"
          />,
          <label
            htmlFor="uploader"
          >
            <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#camera"
              />
            </svg>
          </label>,
        ],
        "className": "profile-image profile-image-uploader centralizer",
        "style": Object {
          "backgroundImage": "linear-gradient(rgba(150, 150, 150, 0.4), rgba(150, 150, 150, 0.4)), url(https://res.cloudinary.com/drmmqcxkc/image/upload/v1541506955/user-placeholder.png)",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "className": "profile-image-uploader_cancel hoverable centralizer",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": "x",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "accept": ".png, .jpg, .jpeg",
            "id": "uploader",
            "onChange": [Function],
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <svg
              className="icon"
            >
              <use
                xlinkHref="icons.svg#camera"
              />
            </svg>,
            "htmlFor": "uploader",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <use
                xlinkHref="icons.svg#camera"
              />,
              "className": "icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "xlinkHref": "icons.svg#camera",
              },
              "ref": null,
              "rendered": null,
              "type": "use",
            },
            "type": "svg",
          },
          "type": "label",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
